### ISSUE TITLE
Add protos and generate gRPC code for Audit Log Datatypes

### ISSUE THREAD

**Issue by bjwatson** [googleapis/googleapis#187]
###What:

Add the proto files for the audit types discussed here: https://cloud.google.com/logging/docs/audit/api/

Generate gRPC code for the appropriate languages:

* [] Python

###Why:

This allows audit log data to be passed as a `proto_payload` field, as documented here: https://cloud.google.com/logging/docs/view/logs_index#audit-logs

**Comment by bjwatson** [comment]
@jba I think you said that this has already been done for Go, correct?

**Comment by bjwatson** [comment]
The AuditLog proto has already been published: https://github.com/googleapis/googleapis/blob/master/google/cloud/audit/audit_log.proto

We just need to make sure it's up to date, since it was last published two months ago.

**Comment by jba** [comment]
Yes, I linked the generated proto
<https://github.com/GoogleCloudPlatform/google-cloud-go/blob/master/logging/logadmin/logadmin.go#L49>
into the Go package that reads logs.

**Comment by bjwatson** [comment]
Awesome, I checked the box for Go in this issue.

**Comment by jba** [comment]
What about the four additional types listed at the bottom of that doc:
```
google.appengine.v1beta4.AuditData
google.appengine.legacy.AuditData
google.cloud.bigquery.logging.v1.AuditData
google.iam.v1.logging.AuditData
```
We should probably include those too, right?

**Comment by bjwatson** [comment]
Yep, good point. We need to publish all of those to https://github.com/googleapis/googleapis

**Comment by bjwatson** [comment]
I updated the comment, and unchecked Go since it doesn't yet have those remaining types.

**Comment by bjwatson** [comment]
Assigning @geigerj for Python part. Other language leads please address your own languages.

FYI, we are planning to release one new package containing both this and the [GAE RequestLog object](https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2572). Please let us know if you think they should be separate packages.

**Comment by bjwatson** [comment]
@nkubala we'll probably fix this for Python at the same time as https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2572.

**Comment by geigerj** [comment]
@bjwatson: @shinfan and I are still working on getting these files open sourced. I'm going to go ahead separate this out from GoogleCloudPlatform/google-cloud-python#2572 in order not to block the other issue. We can add the AuditData types to the package created for that issue when they're open sourced.

**Comment by bjwatson** [comment]
SGTM. BTW, the main motivation for open-sourcing these message types is that they're publicly documented here: https://cloud.google.com/logging/docs/audit/api/#servicedata-services

**Comment by skoushan** [comment]
Any updates on this? I'm specifically looking for the proto of `google.cloud.bigquery.logging.v1.AuditData` described [here](https://cloud.google.com/logging/docs/reference/audit/bigquery/rest/Shared.Types/AuditData).

**Comment by geigerj** [comment]
Hi @skoushan -- thanks for the note. I am taking a look. No ETA currently, unfortunately.

**Comment by skoushan** [comment]
If it's helpful to anyone in the meanwhile, I manually wrote the Protobuf for `google.cloud.bigquery.logging.v1.AuditData` [here](https://gist.github.com/skoushan/c612a5aae19b42a6778f96ba339c18f9).

**Comment by grnadav** [comment]
Any progress on the release of these proto files?
GCP's audit logs are pretty useless without it 😞

**Comment by grnadav** [comment]
@skoushan how were you able to determine the number of each field of a message?
I tried to model IAM's proto but using ordinal order starting at 1 for each message fails to parse the message

**Comment by geigerj** [comment]
@grnadav I'll try to reprioritize this. Is there are particular one of the `AuditData` files you're looking for?

**Comment by grnadav** [comment]
@geigerj by order of importance for me:
google.appengine.v1.AuditData (in orig. post it was google.appengine.v1beta4.AuditData)
google.appengine.legacy.AuditData
google.cloud.bigquery.logging.v1.AuditData  (can use @skoushan 's implemention)
google.iam.v1.logging.AuditData (can use the one I implemented)

**Comment by eoogbe** [comment]
What grpc clients should I generate?

**Comment by grnadav** [comment]
@eoogbe Java is my top priority

**Comment by geigerj** [comment]
@eoogbe The original decision @bjwatson and I came up with was to bundle all of these `AuditData`/`AuditLog`/`RequestLog` types into a new gRPC-only package called `logging-external-types` (sort of like `googleapis-common-protos`) and then set the package metadata of all the GAPIC Stackdriver Logging packages to specify a dependency on `logging-external-types`.

Since @grnadav would benefit most from Java, it SGTM to start there.

**Comment by eoogbe** [comment]
@geigerj When you say gRPC-only package, do you mean a new api?

**Comment by geigerj** [comment]
@eoogbe Well, not an API exactly, since there's no service being defined. Just a new common package that the logging client libraries will depend on.

**Comment by eoogbe** [comment]
@geigerj Are there similar existing packages? Still not clear on what you mean.

**Comment by eoogbe** [comment]
@geigerj Artman can't distinguish between two files named `audit_data.proto`. I don't think the `logging-external-types` package idea will work. Either we can change artman to allow two files with the same name or break up the package into one for each `AuditData`
